pkg/annotations/annotation.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package annotations makes it possible to track use of resource annotations within a procress
+// in order to generate documentation for these uses.
+package annotations
+
+import (
+	"sort"
+	"sync"
+
+	"istio.io/istio/pkg/log"
+)
+
+// Annotation describes a single resource annotation
+type Annotation struct {
```

Comment:
This all seems duplicate of what you've done for env variables. I wonder if there is a way to combine the two? I could see how having typed annotations (e.g. int, bool) might be useful in the same way it is for env variables.  Similarly, env variables can be hidden, deprecated as well.

Reply:
Given the use and context is fairly different, I think distinct implementation with similar patterns is the right choice. It's all tiny code anyway.

Now as it relates to features, do you think it's worth having a standard notion of types for these with canonical parsing/validation logic in pkg/annotations? Looking at what's in inject.go, there's only one generic type used (uint64) and everything else is specialized. WDYT?

Reply:
If we don't want to try to merge the implementations, let's not worry about the typed implementations for now given there wouldn't be much benefit at the moment.